src/OpenLoco/GameCommands/RemoveBuilding.cpp

```diff
@@ -0,0 +1,110 @@
+#include "../CompanyManager.h"
+#include "../Economy/Economy.h"
+#include "../Industry.h"
+#include "../Localisation/FormatArguments.hpp"
+#include "../Localisation/StringIds.h"
+#include "../Map/TileManager.h"
+#include "../Objects/BuildingObject.h"
+#include "../S5/S5.h"
+#include "../TownManager.h"
+#include "GameCommands.h"
+
+namespace OpenLoco::GameCommands
+{
+    // 0x00497D8D
+    static int16_t getCompanyRating(Map::Pos2& pos)
+    {
+        auto companyId = CompanyManager::getUpdatingCompanyId();
+        if (companyId != CompanyId::neutral)
+        {
+            auto res = TownManager::getClosestTownAndUnk(pos);
+            if (res != std::nullopt)
+            {
+                auto townId = res->first;
+                auto town = TownManager::get(townId);
+                if (town != nullptr)
+                {
+                    return town->company_ratings[enumValue(companyId)];
+                }
+            }
+        }
+        return -1;
```

Comment:
This is returning a non negative large number as an indication of failure.
I would change the function to return a std::optional<int16_t>. Then it can return either the company rating or a std::nullopt if it is unsuccessful. The calling code for this particular removeBuilding function would go to the error message only if the value is valid and negative.

Reply:
Good idea! Done.